akka-docs/src/main/paradox/scala/general/actor-systems.md

```diff
@@ -107,14 +107,153 @@ which is a single point of contention if over-used.
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #blocking-in-actor }
+
+When facing this, you
 may be tempted to just wrap the blocking call inside a `Future` and work
 with that instead, but this strategy is too simple: you are quite likely to
 find bottlenecks or run out of memory or threads when the application runs
 under increased load.
 
+@@snip [BlockingDispatcherSample.scalaa](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #blocking-in-future }
+
+
+### Problem
+
+The key here is this line:
+
+```scala
+implicit val executionContext: ExecutionContext = context.dispatcher
+```
+
+Using `context.dispatcher` as the dispatcher on which the blocking Future
+executes can be a problem - the same dispatcher is used for actor's processing,
+unless you @ref:[set up a separate dispatcher for the actor](../dispatchers.md).
+
+If all of the available threads are blocked, then all the actors on the same dispatchers will starve for threads and
+they cannot process any more incoming message.
+
+@@@ note
+
+Blocking APIs should also be avoided if possible. Try to find or build Reactive APIs,
+such that blocking is minimised, or moved over to dedicated dispatchers.
+
+Often when integrating with existing libraries or systems it is not possible to
+avoid blocking APIs. The following solution explains how to handle blocking
+operations properly.
+
+Note that the same hints apply to managing blocking operations anywhere in Akka,
+including in Actors etc.
+
+@@@
+
+### Problem example: blocking the default dispatcher
+
+Let's set up an application with the above `BlockingFutureActor` and the following `PrintActor`. 
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #print-actor }
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #blocking-main }
+
+Here the app is sending 100 messages to `BlockingFutureActor` and `PrintActor` and large numbers
+of akka.actor.default-dispatcher threads are handling requests. When you run run the above code,
+you will likely to see the entire application gets stuck somewhere like this:
+
+```
+>　PrintActor: 44
+>　PrintActor: 45
+```
+
+(i.e.) (Ignoring `println`'s latency) `PrintActor` is non-blocking but it cannot process all the 100 messages asap.
+
+In the thread state diagrams below the colours have the following meaning:
+
+ * Turquoise - Sleeping state
+ * Orange - Waiting state
+ * Green - Runnable state
+
+The thread information was recorded using the YourKit profiler, however any good JVM profiler 
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder). 
+
+The orange portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amounts of Turquoise (sleeping) threads are very bad!
+
```

Comment:
Please add a note here that:

`If you own a Lightbend subscription you can use the commercial [Thread Starvation Detector](http://developer.lightbend.com/docs/akka-commercial-addons/current/starvation-detector.html) which will issue warning log statements if it detects any of your dispatchers suffering from starvation and other. It is a helpful first step to identify the problem is occurring in a production system, and then you can apply the proposed solutions as explained below.`